server/sonar-server/src/main/java/org/sonar/server/issue/workflow/State.java

```diff
@@ -43,47 +44,46 @@ public State(String key, Transition[] outTransitions) {
   }
 
   private static void checkDuplications(Transition[] transitions, String stateKey) {
-    Set<String> keys = Sets.newHashSet();
-    for (Transition transition : transitions) {
-      if (keys.contains(transition.key())) {
+    Set<String> keys = new HashSet<>();
+
+    Arrays.stream(transitions)
+      .filter(transition -> !keys.add(transition.key()))
```

Comment:
I understand. With the lack of accumulator, it's hard to be concise. It's good to use the return value of the `Set.add` method to check for a duplication